### ISSUE TITLE
Overriding plugin settings save action results in SecurityException / Redirect error

### ISSUE THREAD

**Issue by iionly** [Elgg/Elgg#12076]
The problem is at least on Elgg 2.3.7. Not tested on Elgg 3 (which seems to handle it differently).

In actions/plugins/settings/save.php there's the possibility to override the plugin settings save action. Now I'm not sure if there was some recent change causing my current problem (I was quite sure that the overriding worked WITHOUT registering my own save action when I made the last Tidypics release where I first made use of overriding the save action but now it seems necessary to register the action or my own action is never called at all).

Problem is (I think it is) that even when I have my own action to deal with saving the settings, there's a success message added and the forwarding done in any case after the if clause. This seems to clash with any elgg_ok_response / elgg_error_response I have in my own save action.

Could it be that I do something wrong (i.e. do I have to register my own action or not?)? Or has something changed recently? Or would it be a solution to move the system_message() and forward() call in the else part of the if clause? I would need a working solution rather urgently, as it might require that I make a bugfix release of Tidypics and also the next release of izap Videos I'm currently working on is depending on the overriding of the plugin settings save action working without issues on Elgg 2.

**Comment by hypeJunction** [comment]
You always had to register the custom action

**Comment by hypeJunction** [comment]
If you copy paste, make sure you remove the part that checks if action exists, otherwise you run into na infinite loop

**Comment by jeabakker** [comment]
I recently encourtered a similar problem, you can't use `elgg_ok_response` or `elgg_error_response` but need to use the 'old' way of `system_message` / `register_error` and `forward`

**Comment by hypeJunction** [comment]
In 3.x, it's not an issue, because we use the correct action name in the form and not do weird forwarding from the action.

**Comment by iionly** [comment]
I must have messed it up with testing the save action when I made the last Tidypics release. I thought it would work with elgg_ok/error_response but only when not registering the action. I guess I had registered the action at some point and it looked fine when I still had used the old way with the messages and forwarding. But then I didn't notice that the action was not even called anymore when I thought it would also work the new way but without the action registered.

Got it working now. Thanks.
